prune-shared.sh

```diff
@@ -0,0 +1,244 @@
+                    bad_usage "Argument expected: $1"
+                    return 1
+                fi
+                OPT_UPLOAD_HOST=$2
+                shift
+                ;;
+            --upath)
+                if [[ ! ${2:-} ]]; then
+                    bad_usage "Argument expected: $1"
+                    return 1
+                fi
+                OPT_UPLOAD_PATH=$2
+                shift
+                ;;
+            --uuser)
+                if [[ ! ${2:-} ]]; then
+                    bad_usage "Argument expected: $1"
+                    return 1
+                fi
+                OPT_UPLOAD_USER=$2
+                shift
+                ;;
+            --)
+                shift
+                break
+                ;;
+            -*)
+                bad_usage "Unexpected argument: $1"
+                return 1
+                ;;
+            *)
+                positional+=("$1")
+                ;;
+        esac
+        shift
+    done
+
+    positional+=("$@")
+
+    if [[ ${#positional[@]} -ne 1 ]]; then
+        bad_usage "Exactly one positional argument expected"
+        return 1
+    fi
+
+    OPT_SHARED_PATH=${positional[0]}
+
+    if [[ $OPT_SHARED_PATH && $OPT_SHARED_PATH != /* ]]; then
+        OPT_SHARED_PATH=$OPT_UPLOAD_PATH/$OPT_SHARED_PATH
+    fi
+    OPT_SHARED_PATH=$(realpath --canonicalize-missing "$OPT_SHARED_PATH") || return
+}
+
+main()
+{
+    set_defaults || return
+    parse_opts "$@" || return
+
+    if [[ $OPT_H ]]; then
+        short_usage
+        return
+    fi
+
+    if [[ $OPT_HELP ]]; then
+        usage
+        return
+    fi
+
+    local used_images=
+    used_images=$(ssh "$OPT_UPLOAD_USER@$OPT_UPLOAD_HOST" \
+        "find $OPT_UPLOAD_PATH -type l -exec readlink -f {} \\;" \
+        |sort -u |sed -n "s,^$OPT_SHARED_PATH/,,p") || return
+    local all_images=
+    all_images=$(ssh "$OPT_UPLOAD_USER@$OPT_UPLOAD_HOST" "cd $OPT_SHARED_PATH && ls *.7z" |sort) || return
+
+    local unused_images=
+    unused_images=$(join -v 1 <(cat <<<"$all_images") <(cat <<<"$used_images")) || return
+
+    local to_remove=()
+    to_remove=($(join -v 1 <(cat <<<"$unused_images") <(cat <<<"$OPT_KEEP"))) || return
+
+    if [[ ${#to_remove[*]} -eq 0 ]]; then
+        echo "Nothing to be removed" >&2
+    else
+        to_remove+=("${to_remove[@]/%/.meta}" "${to_remove[@]/%/.md5sum}")
+
+        if [[ $OPT_INTERACTIVE ]]; then
+            echo "The following images (and the accompanying files) would be removed from the upload host:"
+            echo
+            (IFS=$'\n'; sed 's/^/\t/' <<<"${to_remove[*]}" |sort)
+            echo
+            local YN=
+            read -p "Continue? [Y/n] " YN
+            [[ $YN && $YN != y ]] && return
+        fi
+
+        local trash=$OPT_SHARED_PATH.TRASH
+        ssh "$OPT_UPLOAD_USER@$OPT_UPLOAD_HOST" \
+            "mkdir -p $trash \
+            && cd $OPT_SHARED_PATH \
+            && mv -v --target-directory=$trash ${to_remove[*]}"
```

Comment:
So the files are just moved to $OPT_SHARED_PATH.TRASH. They are not really removed. Is the idea that the trash directory is later removed manually?

Reply:
yeah, the built-in help is a bit inaccurate (outdated), I will fix it.

Reply:
done